### PR TITLE
Small optimisation for torch.lerp

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91626
* #91625
* #91621

We merge the two `torch.where`s into one.

cc @ezyang @mruberry @ngimel @Lezcano @peterbell10

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -4267,15 +4267,13 @@ def lerp(start: Tensor, end: Tensor, weight: Union[Tensor, NumberType]):
...
-    #    return (1 - weight) * (start - end) + end
-    mask = weight.abs() >= 0.5
-    coeff = torch.where(mask, weight - 1, weight)
-    base = torch.where(mask, end, start)
-    return coeff * (end - start) + base
+    return torch.where(
+        weight.abs() >= 0.5,
+        start + (end - start) * weight,
+        end + (start - end) * (1 - weight),
+    )
```
Comment (peterbell10) [inline torch/_refs/__init__.py:4276]: Have you benchmarked this? There is one fewer where but 2x the number of mathematical operations.

File: torch/_refs/__init__.py [hunk 2]
```diff
@@ -4267,15 +4267,13 @@ def lerp(start: Tensor, end: Tensor, weight: Union[Tensor, NumberType]):
         )
     assert isinstance(weight, Tensor)  # mypy
     # We implement it this way for numerical stability. We assume (in the stability optimisation)
-    # that 0 <= weight <= 1. We take the abs to deal with complex numbers
+    # that -1 <= weight <= 1. We take the abs to deal with complex numbers
     # We want to perform operations near zero, which is where floating points are most precise
-    # thus, we perform the following optimisation:
-    # If weight.abs() >= 0.5:
-    #    return (1 - weight) * (start - end) + end
```
Comment (peterbell10) [inline torch/_refs/__init__.py:4274]: Why get rid of the comment?

Comment (lezcano) [reply torch/_refs/__init__.py:4274]: I just removed it because with the changes this is now obvious.

File: torch/_refs/__init__.py (hunk 1 above)
Comment (lezcano) [reply torch/_refs/__init__.py:4276]: That's fair. Actually what this hints is that I can actually optimise the previous PR. I'll drop this one.

Comment (lezcano) [reply torch/_refs/__init__.py:4276]: Nevermind, this optimisation for lerp, given that a few of these arguments are scalars, does not make sense. You get 5 ops + 1 where vs 3 ops + 2 where. But the where implementation with 2 wheres requires creating potentially large tensors from scalars, which is not great. Leaving it as it is and dropping this PR.

**Pull request closed without merging**